includes/misc-functions.php

```diff
@@ -1363,3 +1995,227 @@ function give_array_insert_after( $key, array &$array, $new_key, $new_value ) {
+				}
+			}
+		}
+
+		return $list;
+	}
+
+	/*
+	 * When index_key is not set for a particular item, push the value
+	 * to the end of the stack. This is how array_column() behaves.
+	 */
+	$newlist = array();
+	foreach ( $list as $value ) {
+		if ( is_object( $value ) ) {
+			if ( isset( $value->$index_key ) ) {
+				$newlist[ $value->$index_key ] = $value->$field;
+			} else {
+				$newlist[] = $value->$field;
+			}
+		} else {
+			if ( isset( $value[ $index_key ] ) ) {
+				$newlist[ $value[ $index_key ] ] = $value[ $field ];
+			} else {
+				$newlist[] = $value[ $field ];
+			}
+		}
+	}
+
+	$list = $newlist;
+
+	return $list;
+}
+
+/**
+ * Get the Import report of the donations
+ *
+ * @since 1.8.13
+ */
+function give_import_donation_report() {
+	return get_option( 'give_import_donation_report', array() );
+}
+
+
+/**
+ * Update the Import report of the donations
+ *
+ * @since 1.8.13
+ */
+function give_import_donation_report_update( $value = array() ) {
+	update_option( 'give_import_donation_report', $value );
+}
+
+
+/**
+ * Delete the Import report of the donations
+ *
+ * @since 1.8.13
+ */
+function give_import_donation_report_reset() {
+	update_option( 'give_import_donation_report', array() );
+}
+
+
+/**
+ * Add meta data field to a customers.
+ *
+ * @since 1.8.13
+ *
+ * @param int $customer_id Customers ID.
+ * @param string $meta_key Metadata name.
+ * @param mixed $meta_value Metadata value. Must be serializable if non-scalar.
+ * @param bool $unique Optional. Whether the same key should not be added.
+ *                           Default false.
+ *
+ * @return int|false Meta ID on success, false on failure.
+ */
+function add_customer_meta( $customer_id, $meta_key, $meta_value, $unique = false ) {
+	return add_metadata( 'give_customer', $customer_id, $meta_key, $meta_value, $unique );
+}
+
+/**
+ * Remove metadata matching criteria from a Customer meta.
+ *
+ * You can match based on the key, or key and value. Removing based on key and
+ * value, will keep from removing duplicate metadata with the same key. It also
+ * allows removing all metadata matching key, if needed.
+ *
+ * @since 1.8.13
+ *
+ * @param int $customer_id Customers ID
+ * @param string $meta_key Metadata name.
+ * @param mixed $meta_value Optional. Metadata value.
+ *
+ * @return bool True on success, false on failure.
+ */
+function delete_customer_meta( $customer_id, $meta_key, $meta_value = '' ) {
+	return delete_metadata( 'give_customer', $customer_id, $meta_key, $meta_value );
+}
+
+/**
+ * Retrieve comment meta field for a Customers.
+ *
+ * @since 1.8.13
+ *
+ * @param int $customer_id Customer ID.
+ * @param string $key Optional. The meta key to retrieve. By default, returns data for all keys.
+ * @param bool $single Whether to return a single value.
+ *
+ * @return mixed Will be an array if $single is false. Will be value of meta data field if $single
+ *  is true.
+ */
+function get_customer_meta( $customer_id, $key = '', $single = false ) {
```

Comment:
@raftaar1191 User donor instead of customer

Reply:
Done